app/Application.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+class Application
+{
+    protected $historyRepository;
+
+    public function __construct()
+    {
+        $DS = DIRECTORY_SEPARATOR;
+
+        require 'constants.php';
+        require "libraries{$DS}console_helpers.php";
+        require "libraries{$DS}helpers.php";
+        require "app{$DS}HistoryRepository.php";
+
+
+        $this->historyRepository = new HistoryRepository();
+    }
+
+    public function run()
+    {
+        $isRunning = true;
+        info_box('', 'Welcome to the calculator app!', '', 'Print "help" to learn more about the app.', '');
+
+        while ($isRunning = true) {
+
+            $command = choose('Enter command: ', AVAILABLE_COMMANDS);
+
+            if (in_array($command, SYSTEM_COMMANDS)) {
+                $this->executeSystemCommand($command, $this->historyRepository);
```

Comment:
you no need to proxy the repository class by methods, you can just use `$this->historyRepositopry` inside the functions when it will be required

```suggestion
                $this->executeSystemCommand($command);
```